### ISSUE TITLE
iOS - How to disable TSan for a third party framework

### ISSUE THREAD

**Issue by KelvinJin** [google/sanitizers#1003]
Hey guys, is there a way to disable TSan for a specific third party framework in iOS?

I know that you can suppress a race/thread.. for a function but what if I want to suppress the entire Framework.

I say Framework here because I know you can disable a .o file or a .c file. But what if I have this .framework file...

**Comment by dvyukov** [comment]
We need to ask Kuba Mracek, but I can't figure out his github username...

**Comment by kubamracek** [comment]
Is the framework is question actually instrumented with TSan instrumentation? If yes, then just skip instrumenting it (don't pass -fsanitize=thread, or if you have a project-wide -fsanitize then use -fno-sanitize=thread on the particular framework), and TSan should not report anything for that framework.

If that doesn't work, please be more specific about what the actual problem is, and how does your project setup look like.

**Comment by KelvinJin** [comment]
@kubamracek Thanks for pointing me to the right direction. It turns out that the framework is not instrumented with TSan. (I checked with `nm xxx.framework/xxx | grep tsan`)

And since `ignore_noninstrumented_modules` is enabled by default on Darwin (which I think is the same for iOS simulator), TSan shouldn't report any false positive. 

I asked the question because I'm having a weird crash related to TSan when running my app. I posted the issue on Swift forum https://forums.swift.org/t/fatal-threadsanitizer-check-failed/16666. I was just trying to make sure this issue is not caused by any 3rd party frameworks I'm using.

**Comment by dvyukov** [comment]
Regardless of instrumentation, false positives and false negatives, tsan must not CHECK-fail. That's a bug in tsan runtime.

The first crash reminds me of:
https://reviews.llvm.org/D45646#1071868
@kubamracek that's still unfixed, right?

Re second crash, we need to see the CHECK failure message, stack alone is not so useful. But either way, it can well be some complex manifestation of the above bug.

**Comment by kubamracek** [comment]
Right, that all sounds right. Can you please file separate bugreports for both these issues, ideally with full crashlogs and stderr outputs, and if possible more info about what triggers the issues.

https://reviews.llvm.org/D45646 indeed is still unfixed.
